dialogs/energy_spectrum.py

```diff
@@ -568,6 +569,8 @@ def __init__(self, parent: BaseTab,
                     self.measurement_energy_spectrum_dictionary
                 sum_spectra_directory = self.simulation.directory
 
+            self.simulation_energy = copy.deepcopy(self.simulation_energy)
+            self.measurement_energy = copy.deepcopy(self.measurement_energy)
             # Graph in matplotlib widget and add to window
             self.matplotlib = MatplotlibEnergySpectrumWidget(
                 self, simulation_energy=self.simulation_energy,
```

Comment:
While this does create a deepcopy, it replaces the original values with it, so the issue of sharing mutable values remains.

Fix:
```suggestion
            widget_simulation_energy = copy.deepcopy(self.simulation_energy)
            widget_measurement_energy = copy.deepcopy(self.measurement_energy)
            # Graph in matplotlib widget and add to window
            self.matplotlib = MatplotlibEnergySpectrumWidget(
                self, simulation_energy=widget_simulation_energy,
                measurement_energy=widget_measurement_energy,
```

(Note that the line 577 isn't highlighted in this change because of [this issue in GitHub](https://github.com/isaacs/github/issues/284))

Reply:
My bad.